Fetch weather data

MassterKek opened the pull request:
Still not functioning as intended

MassterKek committed:
Attempting to fetch weather data

cronin4392 commented:
What issues are you running into? Does something come out of the `console.log`?

cronin4392 commented on js/site.js:
```diff
@@ -1,5 +1,15 @@
 // fetch weather data
 
+  let lat;
```
I think one issue you're gonna have it `lat` and `long` are not set. In the demo I did with the teens we used the browser tool to get your current position. To start though you can just hard code lat and long. I've done this by going on google maps and searching a location. When you do so the URL should update to something like:

```
https://www.google.com/maps/place/Artists+For+Humanity/@42.3429718,-71.0557969,17z/data=!3m2!4b1!5s0x89e37a7cc30fd4d5:0xa53c3af1827a4aa9!4m5!3m4!1s0x89e37a7cc503b901:0xa35db6dcf24f827e!8m2!3d42.3429718!4d-71.0536082
```

The lat and long from that url is `42.3429718` and `-71.0557969`. Please check the weather api documentation for the format they want the lat and long in.

cronin4392 replied:
Their docs are [here](https://www.weatherapi.com/docs/).

MassterKek replied:
I am now getting an object displayed in the console. I assume I should interpret the data from that object and adjust the circles based on that data?

MassterKek committed:
Weather Data retrieval seems to be working.

MassterKek committed:
Color of circle now reflects current temperature.

The hue of the hsl color used for the circle is now based on an input range of 9 to 91 degrees with an output range of 180 (blue) to 360 (red).

cronin4392 commented:
Looking good @MassterKek. For this PR I'd hook up the color to the current temp. Also, you could test by inputting hard coded temps to test a broad range since the temp for the day is gonna be constant. The final code will still use the current temp.

cronin4392 commented on js/site.js:
```diff
@@ -1,18 +1,66 @@
+      mintemp_c,
+    } = data.forecast.forecastday[0].day;
+
+    const hour_0_temp_f = data.forecast.forecastday[0].hour[0].temp_f;
```
I'd remove variables that aren't being used. I think VS Code highlights them?

MassterKek committed:
Cleaned up code and added speed feature

Set animation speed based on current wind speed
